From b66c6facabda35e185956a3418543d4cf7a4255f Mon Sep 17 00:00:00 2001
From: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
Date: Thu, 24 Sep 2020 13:43:20 +0530
Subject: [PATCH 1610/1852] net: xilinx: axiethernet: Pass phy interface to
 of_phy_connect()

commit b9d1ac48ec2d439588e2d8f8f91e8465c005913d from
https://github.com/Xilinx/linux-xlnx.git

Pass phy interface to of_phy_connect() for all the modes (RGMII, GMII,
SGMII,and 1000BaseX). This change makes phy interface a configurable
DT property. It is required to bringup the axiethernet RGMII on kc705
which needs "rgmii-rxid" phy-interface to be passed to the Marvell
(M88E1111) phy driver.

Signed-off-by: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../net/ethernet/xilinx/xilinx_axienet_main.c | 19 ++++---------------
 1 file changed, 4 insertions(+), 15 deletions(-)

diff --git a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
index edaf82091b39..7dca5364f745 100644
--- a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
+++ b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
@@ -1565,21 +1565,10 @@ static int axienet_open(struct net_device *ndev)
 		return ret;
 
 	if (lp->phy_node) {
-		if (lp->phy_mode == XAE_PHY_TYPE_GMII) {
-			phydev = of_phy_connect(lp->ndev, lp->phy_node,
-						axienet_adjust_link, 0,
-						PHY_INTERFACE_MODE_GMII);
-		} else if (lp->phy_mode == XAE_PHY_TYPE_RGMII_2_0) {
-			phydev = of_phy_connect(lp->ndev, lp->phy_node,
-						axienet_adjust_link, 0,
-						PHY_INTERFACE_MODE_RGMII_ID);
-		} else if ((lp->axienet_config->mactype == XAXIENET_1G) ||
-			     (lp->axienet_config->mactype == XAXIENET_2_5G)) {
-			phydev = of_phy_connect(lp->ndev, lp->phy_node,
-						axienet_adjust_link,
-						lp->phy_flags,
-						lp->phy_interface);
-		}
+		phydev = of_phy_connect(lp->ndev, lp->phy_node,
+					axienet_adjust_link,
+					lp->phy_flags,
+					lp->phy_interface);
 
 		if (!phydev)
 			dev_err(lp->dev, "of_phy_connect() failed\n");
-- 
2.31.1

